index_documents.py

```diff
@@ -33,13 +34,17 @@ def split_documents(documents: List[Document], chunk_size: int = 4000, chunk_ove
     text_splitter = RecursiveCharacterTextSplitter(
         chunk_size=chunk_size, chunk_overlap=chunk_overlap)
     # splited_docs = text_splitter.split_documents(documents)
+    
+    # Pattern to remove unsupported hexadecimal characters for Marqo.
+    pattern = re.compile(r'[\x00-\x1F\x7F\u00A0]')
     splited_docs = []
     for document in documents:
         for chunk in text_splitter.split_text(document.text):
+            chunk = re.sub(pattern, '', chunk)
             splited_docs.append(Document(page_content=chunk, metadata={
                 "page_label": document.metadata.get("page_label"),
                 "file_name": document.metadata.get("file_name"),
-                "file_path": document.metadata.get("file_path"),
+                # "file_path": document.metadata.get("file_path"),
```

Comment:
not needed anymore?

Reply:
As we are taking file path from command line argument while indexing so i commented it out as we don't need it anymore